### DISCUSSION TITLE
Plugin for recording video in a trial?

### DISCUSSION THREAD

**Discussion by ShreshthSaxena** [jspsych/jsPsych#2592] [Q&A]
Hi there,
I'm new to jspsych and was looking to create an experiment that records the participant performing the experiment. Ideally we would like to have separate video recordings for each trial, I believe a single recording over the entire experiment might also be problematic in terms of computational requirements and might introduce some lag. 
Is there a plugin already available for this? I came across the webgazer plugins and they seem to do the recording as well however I'm not sure how to access the video recordings afterwards?
If not, how can I start to develop one? any resources would be helpful

Just to be clear, we're aware of the privacy concerns and all our work will be with the GDPR guidelines :)

**Comment by jodeleeuw** [comment]
Hi @ShreshthSaxena,

There's no plugin that records video data yet. The webgazer plugins do use video information but the video data itself is not recorded. It is just analyzed to make gaze predictions.

We have some resources available for building new plugins:

- [Plugin Development Guide](https://www.jspsych.org/7.2/developers/plugin-development/) from the documentation.
- [Video tutorial on building a new plugin](https://www.youtube.com/watch?v=XQcsFwAmbiw) hosted on YouTube. Note that this is for version 6 of jsPsych, but the general approach still applies to version 7.

For information on how to record video I'd look for guides that focus on the `MediaRecorder` element. [Here's one that looks good at first glance](https://api.video/blog/tutorials/building-record-a-video-the-mediarecorder-api).
